File: aat-lib/src/main/java/ch/bailu/aat_lib/service/directory/database/GpxDbConfiguration.java
package ch.bailu.aat_lib.service.directory.database;

public final class GpxDbConfiguration {
    public final static int DB_VERSION = 8;

    public final static String TABLE = "summary";

    public final static String KEY_FILENAME = "filename";
    public final static String KEY_DISTANCE = "distance";
    public final static String KEY_MAX_SPEED = "max_speed";
    public final static String KEY_AVG_SPEED = "avg_speed";
    public static final String KEY_PAUSE = "pause";
    public final static String KEY_TYPE_ID = "type";

    public final static String KEY_EAST_BOUNDING = "east";
    public final static String KEY_WEST_BOUNDING = "west";
    public final static String KEY_NORTH_BOUNDING = "north";
    public final static String KEY_SOUTH_BOUNDING = "south";

    public final static String KEY_ID = "_id";

    public final static String KEY_START_TIME = "start_time";
    public static final String KEY_END_TIME = "end_time";
    public final static String KEY_TOTAL_TIME = "total_time";

    public final static String TYPE_FLOAT = "REAL";
    public final static String TYPE_LONG = "BIGINT";
    public final static String TYPE_STRING = "TEXT";

    public final static String TYPE_ID = "INTEGER PRIMARY KEY AUTOINCREMENT";
    public final static String TYPE_ID_H2 = "INTEGER PRIMARY KEY AUTO_INCREMENT";

    public static final String[] KEY_LIST = new String[]{
            KEY_ID,
            KEY_FILENAME,
            KEY_AVG_SPEED,
            KEY_MAX_SPEED,
            KEY_DISTANCE,
            KEY_START_TIME,
            KEY_TOTAL_TIME,
            KEY_END_TIME,
            KEY_PAUSE,
            KEY_EAST_BOUNDING,
            KEY_WEST_BOUNDING,
            KEY_NORTH_BOUNDING,
            KEY_SOUTH_BOUNDING,
            KEY_TYPE_ID
    };

    public static final String[] TYPE_LIST = new String[]{
            TYPE_ID,
            TYPE_STRING,
            TYPE_FLOAT,
            TYPE_FLOAT,
            TYPE_FLOAT,
            TYPE_LONG,
            TYPE_LONG,
            TYPE_LONG,
            TYPE_LONG,
            TYPE_LONG,
            TYPE_LONG,
            TYPE_LONG,
            TYPE_LONG,
            TYPE_LONG
    };

    public static final String[] TYPE_LIST_H2 = new String[]{
            TYPE_ID_H2,
            TYPE_STRING,
            TYPE_FLOAT,
            TYPE_FLOAT,
            TYPE_FLOAT,
            TYPE_LONG,
            TYPE_LONG,
            TYPE_LONG,
            TYPE_LONG,
            TYPE_LONG,
            TYPE_LONG,
            TYPE_LONG,
            TYPE_LONG,
            TYPE_LONG
    };
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/service/directory/database/GpxDbConfiguration.kt
package ch.bailu.aat_lib.service.directory.database

object GpxDbConfiguration {
    const val DB_VERSION: Int = 8

    const val TABLE: String = "summary"

    const val ATTR_FILENAME: String = "filename"
    const val ATTR_DISTANCE: String = "distance"
    const val ATTR_MAX_SPEED: String = "max_speed"
    const val ATTR_AVG_SPEED: String = "avg_speed"
    const val ATTR_PAUSE: String = "pause"
    const val ATTR_TYPE_ID: String = "type"

    const val ATTR_EAST_BOUNDING: String = "east"
    const val ATTR_WEST_BOUNDING: String = "west"
    const val ATTR_NORTH_BOUNDING: String = "north"
    const val ATTR_SOUTH_BOUNDING: String = "south"

    const val ATTR_ID: String = "_id"

    const val ATTR_START_TIME: String = "start_time"
    const val ATTR_END_TIME: String = "end_time"
    const val ATTR_TOTAL_TIME: String = "total_time"

    private const val TYPE_FLOAT: String = "REAL"
    private const val TYPE_LONG: String = "BIGINT"
    private const val TYPE_STRING: String = "TEXT"

    private const val TYPE_ID: String = "INTEGER PRIMARY KEY AUTOINCREMENT"
    private const val TYPE_ID_H2: String = "INTEGER PRIMARY KEY AUTO_INCREMENT"

    val ATTR_LIST: Array<String> = arrayOf(
        ATTR_ID,
        ATTR_FILENAME,
        ATTR_AVG_SPEED,
        ATTR_MAX_SPEED,
        ATTR_DISTANCE,
        ATTR_START_TIME,
        ATTR_TOTAL_TIME,
        ATTR_END_TIME,
        ATTR_PAUSE,
        ATTR_EAST_BOUNDING,
        ATTR_WEST_BOUNDING,
        ATTR_NORTH_BOUNDING,
        ATTR_SOUTH_BOUNDING,
        ATTR_TYPE_ID
    )

    val TYPE_LIST: Array<String> = arrayOf(
        TYPE_ID,
        TYPE_STRING,
        TYPE_FLOAT,
        TYPE_FLOAT,
        TYPE_FLOAT,
        TYPE_LONG,
        TYPE_LONG,
        TYPE_LONG,
        TYPE_LONG,
        TYPE_LONG,
        TYPE_LONG,
        TYPE_LONG,
        TYPE_LONG,
        TYPE_LONG
    )

    val TYPE_LIST_H2: Array<String> = arrayOf(
        TYPE_ID_H2,
        TYPE_STRING,
        TYPE_FLOAT,
        TYPE_FLOAT,
        TYPE_FLOAT,
        TYPE_LONG,
        TYPE_LONG,
        TYPE_LONG,
        TYPE_LONG,
        TYPE_LONG,
        TYPE_LONG,
        TYPE_LONG,
        TYPE_LONG,
        TYPE_LONG
    )
}
